java-checks/src/main/java/org/sonar/java/checks/ServletMethodsExceptionsThrownCheck.java

```diff
@@ -104,20 +104,43 @@ private void checkMethodInvocation(MethodInvocationTree node) {
     if (symbol.isMethodSymbol()) {
       List<Type> types = ((Symbol.MethodSymbol) symbol).thrownTypes();
       if (!types.isEmpty()) {
-        addIssueIfNotCatched(types, ExpressionUtils.methodName(node), "Add a \"try/catch\" block for \"" + symbol.name() + "\".");
+        addIssueIfNotCaught(types, ExpressionUtils.methodName(node), symbol.name());
       }
     }
   }
 
-  private void addIssueIfNotCatched(Iterable<Type> thrown, Tree node, String message) {
+  private void addIssueIfNotCaught(Type thrown, Tree node) {
+    if (isNotCaught(thrown)) {
+      String message = "Handle the " + "\"" + thrown + "\"" + " thrown here in a \"try/catch\" block.";
```

Comment:
Never rely on the ToString() method of a type. What if it change in implementation class for debugging purpose?
Use `thrown.name()` instead.